### PR TITLE
Move PolycrystalVoronoiVoidIC and Action to MOOSE

### LINKED ISSUE #8721: Move PolycrystalVoronoiVoidIC and PolycrystalVoronoiVoidICAction from Marmot to Moose
### Description of the enhancement or error report
Move PolycrystalVoronoiVoidIC and PolycrystalVoronoiVoidICAction from Marmot to Moose

### Rationale for the enhancement or information for reproducing the error
Allow broader usage of PolycrystalVoronoiVoidIC

### Identified impact
None

### PR DESCRIPTION
**Pull request by laagesen**
Closes #8721 

Removed from Marmot by Marmot MR#246

### CODE REVIEW

File: modules/phase_field/src/ics/PolycrystalVoronoiVoidIC.C
Comment (permcody) [inline modules/phase_field/src/ics/PolycrystalVoronoiVoidIC.C]: All of this assign OPs logic is undergoing changes right now because this "greedy" algorithm doesn't work.
How big of a hurry are we to get this in there? I'd rather not introduce more tests using the soon-to-be "legacy" method. I have an open PR now that adds more advanced coloring options to the various reduced order parameter ICs. However, now that we have a third one, we really should do even more refactoring to further simplify the logic. I'm on travel this week but I can stop in Monday to chat with you about it. Alternatively you can take a look at the "PolycrystalReducedIC" and "ReconVarIC" classes to see what we are doing now. Also see  #8805.

Comment (laagesen) [inline modules/phase_field/src/ics/PolycrystalVoronoiVoidIC.C]: To be honest I'd like to try to get it in to Moose as soon as we can... this is being moved from MARMOT to MOOSE, and the MR to remove it from MARMOT has already been merged. What is driving this is we have been trying to get these objects accessible to an Argonne collaborator who's currently unable to access MARMOT because he is a foreign national and they are not allowing non-INL FN's to get access to INL HPC right now, which he would need for Gitlab. (And apparently as of last week, even INL FN's are on hold for accessing Marmot.) Our collaborator needs these methods to do some simulations of thermal conductivity for the ATF HIP that is supposed to be done this fiscal year, hence the time sensitivity.

Could we merge this as-is for now, and fix it up when your changes to the OP assignment algorithms are done? Since it's already in Marmot like this, it's going to need to be re-worked at some point, whether it's in Marmot or Moose.

**Comment by permcody** [comment]
Yeah, we'll get this merged

**Comment by permcody** [comment]
@laagesen - Merging this now

**Review by permcody — APPROVED** [review]

**Comment by laagesen** [comment]
Thanks @permcody - and if I don't catch it when you finish re-working the OP assignment logic, please let me know and I'll modify this accordingly.

**Pull request merged**
